code/main/common/stateManager.brs

```diff
@@ -0,0 +1,169 @@
+' * you may not use this file except in compliance with the License. You may obtain a copy
+' * of the License at http://www.apache.org/licenses/LICENSE-2.0
+' *
+' * Unless required by applicable law or agreed to in writing, software distributed under
+' * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+' * OF ANY KIND, either express or implied. See the License for the specific language
+' * governing permissions and limitations under the License.
+' *
+' *****************************************************************************************
+
+' ******************************** MODULE: StateManager ***********************************
+
+function _adb_StateManager() as object
+    return {
+        CONFIG_KEY: AdobeSDKConstants().CONFIGURATION,
+        _edge_configId: invalid,
+        _edge_domain: invalid,
+        _ecid: invalid,
+        ' example config = {edge.configId:"1234567890", edge.domain:"xyz.net"}
+        updateConfiguration: function(configuration as object) as void
+            configId = _adb_optStringFromMap(configuration, m.CONFIG_KEY.EDGE_CONFIG_ID)
+            domain = _adb_optStringFromMap(configuration, m.CONFIG_KEY.EDGE_DOMAIN)
+
+            if not _adb_isEmptyOrInvalidString(configId)
+                m._edge_configId = configId
+            end if
+
+            ' example domain: company.data.adobedc.net
+            regexPattern = CreateObject("roRegex", "^((?!-)[A-Za-z0-9-]+(?<!-)\.)+[A-Za-z]{2,6}$", "")
+            if not _adb_isEmptyOrInvalidString(domain) and regexPattern.isMatch(domain)
+                m._edge_domain = domain
+            end if
+        end function,
+
+        resetIdentities: function() as void
+            m.updateECID(invalid)
+        end function,
+
+        getECID: function() as dynamic
+            if m._ecid = invalid
+                _adb_log_verbose("getECID() - ECID not found in cache, fetching it from presistence.")
+                m._ecid = m._loadECID()
+            end if
+
+            if m._ecid = invalid
+                _adb_log_verbose("getECID() - ECID not found in persistence, fetching it from service side.")
+                remote_ecid = m._queryECID()
+                if not _adb_isEmptyOrInvalidString(remote_ecid)
+                    _adb_log_verbose("getECID() - Fetched ECID:(" + FormatJson(m._ecid) + ") from service side")
+                    m.updateECID(remote_ecid)
+                end if
+            end if
+
+            _adb_log_debug("getECID() - Returning ECID:(" + FormatJson(m._ecid) + ")")
+            return m._ecid
+        end function,
+
+        getConfigId: function() as dynamic
+            return m._edge_configId
+        end function,
+
+        getEdgeDomain: function() as dynamic
+            return m._edge_domain
+        end function,
+
+        updateECID: function(ecid as dynamic) as void
+            if ecid = invalid
```

Comment:
let's get changes from this PR https://github.com/adobe/aepsdk-roku/pull/40 before merging.